src/main/java/io/vertx/core/http/impl/NettyFileUploadDataFactory.java

```diff
@@ -46,7 +46,8 @@ public FileUpload createFileUpload(HttpRequest httpRequest, String name, String
       size);
     Handler<HttpServerFileUpload> uploadHandler = lazyUploadHandler.get();
     if (uploadHandler != null) {
-      uploadHandler.handle(upload);
+      // run the handler on the caller context
+      context.runOnContext(v -> uploadHandler.handle(upload));
```

Comment:
instead try `context.execute(upload, uploadHandler)`

Reply:
What's exactly the difference between the two @vietj? I've always used `context.runOnContext` because when I tried to use `context.execute` it would run the handlers outside of the context.

Reply:
Actually it should have be `context.dispatch()`, look at the latest version.

context.execute will execute the task on the context, if the current event loop is already the context then the task is executed directly.